config.TEMPLATE.inc.php

```diff
@@ -282,6 +282,9 @@
 
 [email]
 
+; Path to the sendmail, -bs argument is for using SMTP protocol
+sendmail_path = "/usr/sbin/sendmail -bs"
```

Comment:
@asmecher Vitaliy's work on the email stuff requires some changes to email configuration, to use Laravel's swift mailer. You should probably take a look at this and maybe Vitaliy  can provide some more details about the why/how.

Reply:
Thanks for mentioning this! This is the default configuration recommended by Swift Mailer: https://swiftmailer.symfony.com/docs/sending.html#using-the-sendmail-transport and adopted by Laravel. But when using it, I'm receiving an authentification warning from the sendmail : `X-Authentication-Warning: [host]: [user] owned process doing -bs`. My concern here is if such a warning in headers could increase chances of the mail to be recognized as a spam.

Reply:
@Vitaliy-1, does Swift Mailer always use `sendmail` via command line when delivering via the local server (as opposed to SMTP), or is there an option to use PHP's `mail()`? We're trying to avoid calling `exec()` (https://github.com/pkp/pkp-lib/issues/6077) as  many shared hosts have it disabled.

Reply:
SwiftMailer relies on `proc_*` functions and I don't see anything in the code using `mail()` function in the current master branch. Likely was removed with: https://github.com/swiftmailer/swiftmailer/blob/23e9ffaf6e8c716bce31bde1be708a405b1fbdf9/CHANGES#L106

Reply:
Do you mean a pre-flight check on disabled functions for users who do not have SMTP configured? I think that's a good candidate. However, we shouldn't expect admins to update the configuration file with new parameters *before* they've run the upgrade script, as they are guided to do that with the release notes displayed after the upgrade process, so checking e.g. the sendmail binary being where it's expected shouldn't be part of the pre-flight.

Reply:
> Do you mean a pre-flight check on disabled functions for users who do not have SMTP configured?

Yeah, and other server requirements (eg - PHP version, MySQL version, etc.)

> they are guided to do that with the release notes displayed after the upgrade process

Hmm, shouldn't we add some pre-flight guidance here? If we ever wanted to do things like reset secret keys or send email notifications when upgrades finish/fail, we'd probably want to be able to ensure that required configuration parameters are in place, right?

Reply:
> If we ever wanted to do things like reset secret keys or...

While we haven't done that kind of thing before, I agree that the examples you put forward would be best done as pre-flight checks. However, I think upgrade steps that involve irreversible interactions (like communicating with third-party services or sending emails) should be avoided, because a lot of folks will want to run a test upgrade before committing to a production upgrade.

Reply:
@asmecher are you happy for this to be merged and we can file an issue against 3.4 to address the issues raised by this new requirement?

Reply:
(if the user forgot to update config file, default Swift Mailer's setting on sendmail path will be used)

Reply:
Yes, please! Thanks to you both.

Reply:
I've filed an issue at https://github.com/pkp/pkp-lib/issues/7285.